components/misc/point.h

```diff
@@ -0,0 +1,35 @@
+#pragma once
+
+#include <cinttypes>
+#include <functional>
+
+namespace Misc
+{
+    struct Point
+    {
+        int32_t x;
+        int32_t y;
+
+        Point();
+        Point(int32_t x, int32_t y);
+
+        Point operator+(const Point& v) const { return {x + v.x, y + v.y}; }
+        Point operator-(const Point& v) const { return {x - v.x, y - v.y}; }
+        Point operator*(double c) const { return {static_cast<int32_t>(x * c), static_cast<int32_t>(y * c)}; }
```

Comment:
This should probably be a FixedPoint, not a double.
Not sure where it's used, but if we really need a double version (for rendering, we should never use it for the actual game logic), we can just do it inline.

Reply:
They are used in sdl2backend.cpp
I've marked this lines as "inline"